bundles/org.openhab.binding.viessmann/src/main/java/org/openhab/binding/viessmann/internal/handler/DeviceHandler.java

```diff
@@ -0,0 +1,1165 @@
+                newChannelType = "slope";
+                updateChannelType = true;
+            }
+
+            if ("shift".equals(oldId) && oldChannelType.contains("number")) {
+                newChannelType = "shift";
+                updateChannelType = true;
+            }
+
+            if (!newId.equals(oldId) || updateChannelType) {
+                logger.info("Migrating channel '{}' -> '{}' | channel-type  '{}' -> '{}'", oldId, newId, oldChannelType,
+                        newChannelType);
+
+                ChannelUID oldUid = channel.getUID();
+                ChannelUID newUid = new ChannelUID(thing.getUID(), newId);
+                String channelDescription = channel.getDescription();
+                if (channelDescription == null) {
+                    channelDescription = "";
+                }
+
+                ChannelTypeUID channelTypeUID = new ChannelTypeUID(BINDING_ID, newChannelType);
+                if (channelLabel != null) {
+                    Channel newChannel = ChannelBuilder.create(newUid, channel.getAcceptedItemType())
+                            .withLabel(channelLabel).withDescription(channelDescription).withType(channelTypeUID)
+                            .withProperties(channel.getProperties()).build();
+
+                    newChannels.add(newChannel);
+                    renameMap.put(oldUid, newUid);
+                }
+            }
+        }
+
+        if (renameMap.isEmpty()) {
+            return;
+        }
+
+        updateThing(editThing().withChannels(newChannels).build());
+
+        if (linkRegistry != null) {
+            for (Map.Entry<ChannelUID, ChannelUID> e : renameMap.entrySet()) {
+                ChannelUID oldUid = e.getKey();
+                ChannelUID newUid = e.getValue();
+
+                Collection<ItemChannelLink> links = new ArrayList<>(linkRegistry.getLinks(oldUid));
+
+                for (ItemChannelLink link : links) {
+                    String item = link.getItemName();
+                    try {
+                        linkRegistry.remove(link.getUID());
+                    } catch (Exception ex) {
+                        logger.warn("Could not remove old link {} -> {}: {}", item, oldUid, ex.getMessage());
+                    }
+
+                    linkRegistry.add(new ItemChannelLink(item, newUid));
+                    logger.info("Re-linked item '{}' from '{}' to '{}'", item, oldUid.getId(), newUid.getId());
+                }
+            }
+        } else {
+            logger.warn("ItemChannelLinkRegistry not available – cannot migrate item links.");
+        }
+    }
+
+    @Override
+    public void initChannelState() {
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            BridgeHandler bridgeHandler = bridge.getHandler();
+            if (bridgeHandler != null) {
+                if (bridgeHandler instanceof BridgeInterface bridgeInterface) {
```

Comment:
This is used ad multiple places, can be simplified and extracted. 
```suggestion
        private @Nullable BridgeInterface getBridgeInterface() {
                if (getBridge() instanceof Bridge bridge) {
                        if (bridge.getHandler() instanceof BridgeInterface bridgeInterface) {
                                return bridgeInterface;
                        }
                }
                return null;
        }   
```